test/Middleware/ImplicitHeadMiddlewareTest.php

```diff
@@ -157,4 +158,47 @@ public function testInvokesHandlerWhenRouteImplicitlySupportsHeadAndSupportsGet(
 
         $this->assertSame($response->reveal(), $result);
     }
+
+    public function testInvokesHandlerWithRequestComposingRouteResultAndAttributes()
+    {
+        $result = $this->prophesize(RouteResult::class);
+        $result->getMatchedRoute()->willReturn(false);
+
+        $request = $this->prophesize(ServerRequestInterface::class);
+        $request->getMethod()->willReturn(RequestMethod::METHOD_HEAD);
+        $request->getAttribute(RouteResult::class)->will([$result, 'reveal']);
+        $request->withMethod(RequestMethod::METHOD_GET)->will([$request, 'reveal']);
+        $request
+            ->withAttribute(
+                ImplicitHeadMiddleware::FORWARDED_HTTP_METHOD_ATTRIBUTE,
+                RequestMethod::METHOD_HEAD
+            )
+            ->will([$request, 'reveal']);
+
+        $response = $this->prophesize(ResponseInterface::class);
+        $response->withBody($this->stream->reveal())->will([$response, 'reveal']);
+
+        $route = $this->prophesize(Route::class);
+
+        $parameters = ['foo' => 'bar', 'baz' => 'bat'];
+        $result = $this->prophesize(RouteResult::class);
+        $result->isFailure()->willReturn(false);
```

Comment:
As commented above. Here you are mocking it because you know what are the internal calls (because you checked them). Why you are not mocking `isSuccess` method? I think this test is not future proof, so as I suggested above RouteResult should NOT be mocked.